sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -397,17 +398,21 @@ public String getEventHubName() {
             ? new AtomicReference<>(Context.NONE)
             : null;
 
+        // initiate addLink on send span, get shared span builder for all links
+        Context sharedContext = isTracingEnabled ? tracerProvider.startSpan(parentContext.get(), ProcessKind.LINK) :
```

Comment:
does it really starts a span? we'll start another one somewhere later - why do we need two?

Reply:
no this, doesn't it just returns a span builder. I wanted to stay consistent with the exposed methods that are usually used around with tracing, so used an existing one.

Reply:
no, it will just return a span builder created with "Azure.eventhubs.send" name. Just wanted to keep the naming consistent with the other methods and so used an existing one.

Reply:
I see. I understand the intention to stay in the existing public API boundaries, but  it seems to be very misleading. startSpan sometimes start span, sometimes just creates a builder in the context: it's completely custom behavior for each kind. It seems very hard to read and maintain. Also I believe it makes almost impossible for someone to implement tracer - they need to understand all the rules hidden in 

https://github.com/Azure/azure-sdk-for-java/blob/9076e2f7bb141d9be4ad9c788c5fa7c4cf12795c/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryTracer.java#L67..L88

While it's surely up to you how to proceed, I suggest to split it into a different method. Or at least leave a comment/hints that explain what's going on here.

Reply:
should we rather just add a new method `getSpanBuilder()` instead?